src/columnsTypes/types/Date/Component.jsx

```diff
@@ -69,6 +69,10 @@ export const DatePickerComponent = ({view, node, pos}) => {
   const ref = useClickOutside((e) => {
     if (!view.dom.contains(e.target)) return;
 
+    // Skip click events, only listen for Esc events.
+    // See https://linear.app/skiff/issue/ENG-3320/table-date-picker-breaks-if-you-click-directly-from-one-cell-to
+    if (e.type === 'click') return;
```

Comment:
In that case maybe it will make sense to remove the `useClickOutside` and create a listener for Esc only